### PR TITLE
Full code refactor

### PR DESCRIPTION
**Pull request by trufae**
**Detailed description**

...

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

File: r2env/config/profiles.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    "versions": [
+      "latest"
+    ]
+  },
+  "radare2": {
+    "source": "https://github.com/radareorg/radare2",
+    "versions": [
+      "5.0.0",
+      "5.1.0",
+      "5.2.1",
```
Comment (trufae) [inline r2env/config/profiles.json:13]:
```suggestion
      "5.2.1",
      "5.3.0",
```

File: r2env/config/r0.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    try:
+        os.mkdir(dstdir)
+    except Exception:
+        pass
+    try:
+        os.mkdir(logdir)
+    except Exception:
+        pass
+    if os.path.isdir(gitdir):
+        rc = os.system("cd '" + gitdir + "' && git reset --hard ; git checkout master ; git reset --hard && git pull")
```
Comment (trufae) [inline r2env/config/r0.py:34]:
```suggestion
        rc = os.system("cd '" + gitdir + "' && git reset --hard ; git checkout master ; git pull")
```

File: r2env/config/r0.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    else:
+        rc = os.system("git clone https://github.com/radareorg/r0 '" + gitdir + "'")
+    if rc != 0:
+        print("Clone failed")
```
Comment (trufae) [inline r2env/config/r0.py:38]:
```suggestion
        print("Clone or pull failed")
```

File: r2env/config/r0.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        return False
+    if "tag" in profile:
+        os.system("git checkout " + str(profile["tag"]))
+    print("Building ...")
+    print("tail -f " + logfil)
+    rc = os.system("(cd " + gitdir + " &&  git clean -xdf && make -j4 PREFIX=" + prefix +
+                   " 2>&1 && make install PREFIX=" + prefix + " DESTDIR=" + dstdir + ")")  # > " + logfil)
+    if rc == 0:
+        # clone r0 in srcdir
+        os.system("date > " + dstdir + "/.timestamp.txt")
```
Comment (trufae) [inline r2env/config/r0.py:48]: this is not portable, should be rewritten in python, well, most of the os.sytem() calls wont work on windows :D just a random reminder

File: r2env/config/r0.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    else:
+        rc = os.system("git clone https://github.com/radareorg/r0 '" + gitdir + "'")
+    if rc != 0:
+        print("Clone failed")
+        return False
+    if "tag" in profile:
+        os.system("git checkout " + str(profile["tag"]))
+    print("Building ...")
+    print("tail -f " + logfil)
+    rc = os.system("(cd " + gitdir + " &&  git clean -xdf && make -j4 PREFIX=" + prefix +
```
Comment (trufae) [inline r2env/config/r0.py:44]: use python instead of os.system(). this will fail if gitdir contain spaces

File: r2env/config/radare2.py [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+    print("Building ...")
+    print("tail -f " + logfil)
+    use_meson = profile["meson"]
+    if use_meson:
+        rc = os.system(
+            "(cd " + gitdir + " && git clean -xdf && rm -rf build && meson --buildtype=release --prefix=" + prefix + " 2>&1 && ninja -C build && ninja -C build install DESTDIR=" + dstdir + ") > " + logfil)
+    else:
+        # rc = os.system("(cd " + gitdir + " && git clean -xdf && rm -rf shlr/capstone; ./configure --with-rpath --prefix="+dstdir+" 2>&1 && make -j4 2>&1 && make install DESTDIR="+dstdir+") > " + logfil)
+        rc = os.system(
+            "(cd " + gitdir + " && git clean -xdf && rm -rf shlr/capstone; ./configure --prefix=" + dstdir + " 2>&1 && make -j4 2>&1 && make install DESTDIR=" + dstdir + ") > " + logfil)
```
Comment (trufae) [inline r2env/config/radare2.py:52]:
```suggestion
            "(cd " + gitdir + " && git clean -xdf && rm -rf shlr/capstone; ./configure --with-rpath --prefix=" + dstdir + " 2>&1 && make -j4 2>&1 && make install) > " + logfil)
```

File: r2env/config/radare2.py [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+        print("Clone failed")
+        return False
+    if "tag" in profile:
+        os.system("cd " + gitdir + " && git checkout " + str(profile["tag"]))
+    print("Building ...")
+    print("tail -f " + logfil)
+    use_meson = profile["meson"]
+    if use_meson:
+        rc = os.system(
+            "(cd " + gitdir + " && git clean -xdf && rm -rf build && meson --buildtype=release --prefix=" + prefix + " 2>&1 && ninja -C build && ninja -C build install DESTDIR=" + dstdir + ") > " + logfil)
```
Comment (trufae) [inline r2env/config/radare2.py:48]:
```suggestion
            "(cd " + gitdir + " && git clean -xdf && rm -rf build && meson --buildtype=release local=true --prefix=" + prefix + " 2>&1 && ninja -C build && ninja -C build install) > " + logfil)
```

File: r2env/config/radare2.py [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+        rc = os.system(
+            "(cd " + gitdir + " && git clean -xdf && rm -rf shlr/capstone; ./configure --prefix=" + dstdir + " 2>&1 && make -j4 2>&1 && make install DESTDIR=" + dstdir + ") > " + logfil)
+    if rc == 0:
+        # clone radare2 in srcdir
+        os.system("date > " + dstdir + "/.timestamp.txt")
+        os.system("git log |head -n1> " + dstdir + "/.commit.txt")
+    else:
+        print("Build failed")
+        sys.exit(1)
+
```
Comment (trufae) [inline r2env/config/radare2.py:60]:
```suggestion
```

File: r2env/config/radare2.py [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+    else:
+        # rc = os.system("(cd " + gitdir + " && git clean -xdf && rm -rf shlr/capstone; ./configure --with-rpath --prefix="+dstdir+" 2>&1 && make -j4 2>&1 && make install DESTDIR="+dstdir+") > " + logfil)
+        rc = os.system(
+            "(cd " + gitdir + " && git clean -xdf && rm -rf shlr/capstone; ./configure --prefix=" + dstdir + " 2>&1 && make -j4 2>&1 && make install DESTDIR=" + dstdir + ") > " + logfil)
+    if rc == 0:
+        # clone radare2 in srcdir
+        os.system("date > " + dstdir + "/.timestamp.txt")
+        os.system("git log |head -n1> " + dstdir + "/.commit.txt")
+    else:
+        print("Build failed")
```
Comment (trufae) [inline r2env/config/radare2.py:58]: stderr

File: r2env/repl.py [hunk 10]
```diff
@@ -1,182 +1,75 @@
...
-	pkg.build(profile)
-	pkg.install()
-
-def del_package(pkg, profile):
-	print("Deleting package")
-	pkg.clean(profile)
 
-def cb_use(sauce, prefix):
-	dploy.unstow([sauce], prefix)
-	dploy.stow(  [sauce], prefix)
```
Comment (trufae) [inline r2env/repl.py:58]: use dploy pls :P

Comment (as0ler) [reply r2env/repl.py:58]: I'll do it, but in another place ;)

**Pull request merged**